internal/database/db.go

```diff
@@ -23,20 +25,23 @@ func NewDB(stats *stats.Stats) *DB {
 
 // Load loads a number from a fake database.
 // Normally, we would propagate the trace ID to the database.
-func (db *DB) Load(ctx context.Context) int {
-	ctx, span := beeline.StartSpan(ctx, "loading")
+func (db *DB) Query(ctx context.Context, name string, n int) (int, error) {
+	if util.Lucky(0.1) {
+		return 0, errors.New("query." + name)
+	}
+
+	ctx, span := beeline.StartSpan(ctx, "query")
```

Comment:
I would create the span even when the query fails (i.e. move span creation before the "if util.Lucky(0.1)"), what do you think?

Reply:
i have added as well the error in the span in this case